Automattic/delphin
# Checkout: Payment information is not cleared when user leaves the page

Fixes #1085:

> ### How to reproduce
> - Complete the checkout flow, (use a fake cc and do not sandbox so your purchase fails)
> - Click on the get.blog logo and restart the flow with another domain
> - Notice that your Payment Information is still there, even if you left the page several hours ago
> 
> ### Suggested action
> We should clear the Checkout form when the user leaves the page

I found it easier to always clear checkout data when user enters page. Though now I think we could also put this logic in `componentWillUnmount`. Not sure what is better. I didn't play with the latter.

### After changes
![checkout-transaction-reset](https://cloud.githubusercontent.com/assets/699132/21453654/922c1e1a-c911-11e6-9d0e-642260519ba6.gif)

### Testing
1. Open http://delphin.live:1337 or https://delphin.live/?branch=fix/1085-payment-error-clear.
2. Complete the checkout flow, (use a fake cc and do not sandbox so your purchase fails)
3. Click on the get.blog logo and restart the flow with another domain
4. Make sure payment error information is no longer there.

### Review

@Automattic/sdev-feed
